toptal/picasso
# feat: add heading

[FX-4008]

### Description

This pull request adds support for normal and heading (`<h3/>`) text in Rich Text Editor.

### How to test

- check out the branch locally, run `yarn` and `yarn start`
- copy and paste text with heading from https://github.com/facebook/lexical#editor-states
- compare behavior with https://picasso.toptal.net/?path=/story/forms-richtexteditor--richtexteditor
- copy and paste text with multiple types of headings from https://docs.google.com/document/d/1uiVxvvkfGfCb7SaW_4tfKLl4sxpSwyaPra-4SPQR6iA/edit?usp=sharing, non-`h3` headings should be replaced with regular bold text

### Screenshots

https://github.com/toptal/picasso/assets/1390758/533fa6df-e0f9-4ab9-972e-739b258d5798

---

https://github.com/toptal/picasso/assets/1390758/23105f68-9e28-4da2-929f-8f69dff421fc

### Development checks

- [N/A] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [N/A] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [N/A] Covered with tests

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-4008]: https://toptal-core.atlassian.net/browse/FX-4008?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
